src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -196,10 +198,41 @@ public void deleteAttachment(String requestId, String companyNumber, String obje
                 () -> new AttachmentNotFoundException(String.format(ATTACHMENT_NOT_FOUND_MESSAGE, attachmentId))
         );
 
-        attachments.remove(attachment);
+        boolean successfulDelete = deleteFromS3(requestId, companyNumber, objectionId, attachmentId);
 
-        objection.setAttachments(attachments);
+        if(successfulDelete) {
```

Comment:
if delete is unsuccessful we should throw an exception so that the controller can return a 500. As it is an unsuccessful deletion would return a 204 NO CONTENT